lib/promotion.rb

```diff
@@ -134,14 +134,19 @@ def self.tl3_lost?(user)
   def self.recalculate(user, performed_by = nil, use_previous_trust_level: false)
     granted_trust_level =
       TrustLevel.calculate(user, use_previous_trust_level: use_previous_trust_level)
-    return user.update(trust_level: granted_trust_level) if granted_trust_level.present?
 
-    user.update_column(:trust_level, TrustLevel[0])
+    if user.manual_locked_trust_level.present?
+      return user.update(trust_level: granted_trust_level) if granted_trust_level.present?
+    end
+
+    user.update_column(:trust_level, TrustLevel[granted_trust_level])
```

Comment:
This is a bit weird since it's doing the same thing as above?  `user.update(trust_level: granted_trust_level)` vs `user.update_column(:trust_level, TrustLevel[granted_trust_level])`. Seems like we just unconditionally should `user.update!(trust_level: TrustLevel[granted_trust_level])` and only do the early return if ` user.manual_locked_trust_level.present?`

Reply:
ahh yeah will fix